eslint/eslint-plugin/src/typedef-var.test.ts

```diff
@@ -6,7 +6,8 @@ import { typedefVar } from './typedef-var';
 
 const { RuleTester } = ESLintUtils;
 const ruleTester = new RuleTester({
-  parser: '@typescript-eslint/parser'
+  // In PNPN 7.0 using '@typescript-eslint/parser' unexpectedly fails, se we are resolving the absolute path of the parser
```

Comment:
```suggestion
  /*
   * The underlying API requires an absolute path. `@typescript-eslint/utils` calls `require.resolve()` on the input
   * and forces it to be of type '@typescript-eslint/parser' but does not have a dependency on `@typescript-eslint/parser`
   * This means that it will always fail to resolve in a strict environment.
   * Fortunately `require.resolve(absolutePath)` returns `absolutePath`, so we can resolve it first and cast.
   */
```